games-misc/cowsay/cowsay-3.03.recipe

```diff
@@ -0,0 +1,43 @@
+SUMMARY="Configurable speaking/thinking cow (and a bit more)"
+DESCRIPTION="Configurable talking ASCII cow and other characters. \
+Invoking it as cowthink will make it think instead of speak."
+HOMEPAGE="https://github.com/tnalpgge/rank-amateur-cowsay"
+COPYRIGHT="1999 Tony Monroe"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="$HOMEPAGE/archive/$portName-$portVersion.tar.gz"
+CHECKSUM_SHA256="9a32cba0b561719e1302398857124fdf4dd21a7623c017eaadbf31277f9274af"
+SOURCE_DIR="rank-amateur-cowsay-cowsay-3.03"
+
+ARCHITECTURES="any"
+
+PROVIDES="
+	$portName = $portVersion
+	cmd:cowsay
+	cmd:cowthink
+	"
+REQUIRES="
+	haiku
+	perl
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+
+PATCH()
+{
+	sed -i -e 's,%BANGPERL%,!/bin/perl,' \
+	-e "s,[^[:space:]\']*/share/cows,/boot/system/data/cows," cowsay{,.1}
```

Comment:
I agree; I just wasn’t sure if it fitted in `INSTALL`. Is there any reason why you patched them after installation and used a glob after `README` (habit to match `.md`?)? Also, I don’t think `/usr` exists in Haiku. ;) Thanks for taking the time to give all these suggestions!

Reply:
Yep, for this particular case, patching the script after copying it to `$binDir` seems more appropriate because what we're doing with the sed command is just replacing the shebang and adjusting the $dataDir. It's not something that would be up-streamable.
For the README files, I could have used `README README.md` but since `README*` expands to these exact files, I think it is nicer to use the glob :)
Regarding the `#!/usr/bin/env perl`, you are right: Haiku does not have a /usr/ directory. But the system has a special test which checks for the presence of a `#!/usr/bin/env` followed by the name of a runtime. If it finds this string then it behaves as expected: it looks for the runtime (perl in our case) in the PATH. This is, imho, slightly better than the `/bin/perl` because it allows cowsay to use whatever perl the user has installed. That said, `/bin/perl` is also fine for me.

Reply:
There isn’t any `README.md` as far as I can see, though. Thanks for the explanations!